File: agentwallet/packages/sdk-python/src/agentwallet/resources/agents.py
"""Agents sub-resource."""

from __future__ import annotations

from typing import TYPE_CHECKING

from ..types import Agent, ListResponse

if TYPE_CHECKING:
    from ..client import AgentWallet


class AgentsResource:
    def __init__(self, client: AgentWallet):
        self._client = client

    async def create(
        self,
        name: str,
        description: str | None = None,
        capabilities: list[str] | None = None,
        is_public: bool = False,
        metadata: dict | None = None,
    ) -> Agent:
        data = await self._client.post("/agents", json={
            "name": name,
            "description": description,
            "capabilities": capabilities or [],
            "is_public": is_public,
            "metadata": metadata or {},
        })
        return Agent(**data)

    async def get(self, agent_id: str) -> Agent:
        data = await self._client.get(f"/agents/{agent_id}")
        return Agent(**data)

    async def list(
        self,
        status: str | None = None,
        limit: int = 50,
        offset: int = 0,
    ) -> ListResponse:
        params = {"limit": limit, "offset": offset}
        if status:
            params["status"] = status
        data = await self._client.get("/agents", params=params)
        return ListResponse(
            data=[Agent(**a) for a in data["data"]],
            total=data["total"],
        )

    async def update(self, agent_id: str, **kwargs) -> Agent:
        data = await self._client.patch(f"/agents/{agent_id}", json=kwargs)
        return Agent(**data)

File: agentwallet/packages/sdk-ts/src/resources/agents.ts
import type { AgentWallet } from '../client';
import type { Agent, CreateAgentParams, ListResponse } from '../types';

export class AgentsResource {
  constructor(private client: AgentWallet) {}

  /** Create a new AI agent */
  async create(params: CreateAgentParams): Promise<Agent> {
    return this.client.post<Agent>('/agents', {
      name: params.name,
      description: params.description ?? null,
      capabilities: params.capabilities ?? [],
      is_public: params.is_public ?? false,
      metadata: params.metadata ?? {},
    });
  }

  /** Get an agent by ID */
  async get(agentId: string): Promise<Agent> {
    return this.client.get<Agent>(`/agents/${agentId}`);
  }

  /** List agents */
  async list(options?: { status?: string; limit?: number; offset?: number }): Promise<ListResponse<Agent>> {
    return this.client.get<ListResponse<Agent>>('/agents', {
      status: options?.status,
      limit: options?.limit ?? 50,
      offset: options?.offset ?? 0,
    });
  }

  /** Update an agent */
  async update(agentId: string, params: Partial<CreateAgentParams>): Promise<Agent> {
    return this.client.patch<Agent>(`/agents/${agentId}`, params as Record<string, unknown>);
  }
}
